plugins/modules/vmware_host_logbundle.py

```diff
@@ -0,0 +1,245 @@
+        - VirtualMachines:VirtualMachineStats
+        - VirtualMachines:base
+        - VirtualMachines:base
+        - VirtualMachines:diskinfo
+        - VirtualMachines:logs
+        - Storage:FCoE
+        - Storage:Multipathing
+        - Storage:NAS
+        - Storage:VSAN
+        - Storage:VSANHealth
+        - Storage:VSANIscsiTarget
+        - Storage:VSANPerfStats
+        - Storage:VSANPerfSvc
+        - Storage:VSANTraces
+        - Storage:VVOL
+        - Storage:base
+        - Storage:iodm
+        - Storage:iscsi
+        - FeatureStateSwitch:FeatureStateSwitch
+        - Userworld:HostAgent
+        - Userworld:ProcessInformation
+        - Configuration:System
+        - Logs:System
+        - hostProfiles:SystemImageCacheHostProfile
+        - hostProfiles:hostProfiles
+        - FileSystem:VMFSDiskDump
+        - FileSystem:base
+        - ActiveDirectory:base
+        - CIM:base
+        - Hardware:base
+        - Hardware:usb
+        - Installer:base
+        - Network:base
+        - Network:dvs
+        - Network:lacp
+        - Network:nscd
+        - Network:tcpip
+        - IntegrityChecks:md5sums
+      type: list
+      elements: str
+      required: False
+extends_documentation_fragment:
+    - community.vmware.vmware.documentation
+'''
+
+EXAMPLES = '''
+- name: fetch logbundle file from ESXi
+  vmware_host_logbundle:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    esxi_hostname: "{{ esxi_hostname }}"
+    dest: ./esxi-log.tgz
+
+- name: fetch logbundle file from ESXi with manifests
+  vmware_host_logbundle:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    validate_certs: no
+    esxi_hostname: "{{ esxi_hostname }}"
+    dest: ./esxi-log.tgz
+    manifests:
+      - System:Base
+      - VirtualMachines:VirtualMachineStats
+'''
+
+RETURN = '''
+dest:
+    description: saved path of a logbundle file for ESXi
+    returned: on success
+    type: str
+    sample:
+      {
+        "changed": true,
+        "dest": "./esxi-log.tgz",
+        "failed": false,
+        "gid": 0,
+        "group": "root",
+        "mode": "0644",
+        "owner": "root",
+        "size": 25783140,
+        "state": "file",
+        "uid": 0
+      }
+'''
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible_collections.community.vmware.plugins.module_utils.vmware import PyVmomi, vmware_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+import xml.etree.ElementTree as ET
+
+
+class VMwareHostLogbundle(PyVmomi):
+    def __init__(self, module):
+        super(VMwareHostLogbundle, self).__init__(module)
+        self.esxi_hostname = self.params['esxi_hostname']
+        self.dest = self.params['dest']
+        self.manifests = self.params['manifests']
+
```

Comment:
Add a check here if the user runs this module on vCenter? Not a blocker though.

Reply:
> Add a check here if the user runs this module on vCenter?

Could you please tell me if there is a useful code for the reference? (or code implemented in other modules.)

Reply:
@Akasurde

I've checked this module on vCenter and it's fine, so I'm assuming there's no need to check it here.
What do you think?

Reply:
Thanks for the test, it should be just all right.